Add DEVICE_SERIAL_FC guards to serial HAL API

fkjagodzinski opened the pull request:
### Description

<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->

Functions related to serial flow control should not be exposed if `DEVICE_SERIAL_FC` is not defined
* `serial_set_flow_control`,
* `serial_cts_pinmap`,
* `serial_rts_pinmap`.

The `tests-mbed_hal_fpga_ci_test_shield-uart` is also updated here.
This patch should fix build errors found here: https://github.com/ARMmbed/mbed-os/pull/10965#issuecomment-509369593.

### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

<!--
    Optional
    Request additional reviewers with @username
-->

@0xc0170 @mprse 

### Release Notes

<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

fkjagodzinski committed:
HAL: Serial: Add DEVICE_SERIAL_FC guards

Functions related to serial flow control should not be exposed
if DEVICE_SERIAL_FC is not defined
* serial_set_flow_control,
* serial_cts_pinmap,
* serial_rts_pinmap.

fkjagodzinski committed:
Test: HAL: serial: Add DEVICE_SERIAL_FC guards

Fix undefined references to serial_set_flow_control.

jamesbeyond reviewed (approved):
LGTM

mprse commented on hal/serial_api.h:
```diff
@@ -211,6 +211,7 @@ void serial_break_clear(serial_t *obj);
  */
 void serial_pinout_tx(PinName tx);
 
+#if DEVICE_SERIAL_FC
```
I think that this commit should be removed for now. 
It will probably cause build failure at least here:
https://github.com/ARMmbed/mbed-os/blob/f0ec856020290a2309d713ae8d9b1d51cab7fbe3/targets/TARGET_RDA/TARGET_UNO_91H/serial_api.c#L131-L133

fkjagodzinski replied:
Yeah, the build failed as expected. This won't fail after https://github.com/ARMmbed/mbed-os/pull/10994/commits/4842c87815c64401b92a3871b8f79e8f559384a2. Could you re-review, please?

fkjagodzinski committed:
FVP_MPS2: Fix serial_init when FC is not used

After adding DEVICE_SERIAL_FC guards to serial_api.h
serial_set_flow_control is not available. In case of this
implementation, this function is a no-op and may be safely removed.

fkjagodzinski committed:
MPS2: Fix serial_init when FC is not used

After adding DEVICE_SERIAL_FC guards to serial_api.h
serial_set_flow_control is not available. In case of this
implementation, this function is a no-op and may be safely removed.

fkjagodzinski committed:
RDA5981X: Add SERIAL_FC

According to comment in the implementation this target supports the
hardware flow control on UART1 peripheral.
This patch fixes build errors after adding DEVICE_SERIAL_FC guards to
hal/serial_api.h.
